maplibre/src/style/style.rs

```diff
@@ -16,117 +20,243 @@ use crate::style::{
 pub struct Style {
     pub version: u16,
     pub name: String,
-    pub metadata: HashMap<String, String>,
-    pub sources: HashMap<String, Source>,
-    pub layers: Vec<StyleLayer>,
-    pub center: Option<[f64; 2]>, // TODO: Use LatLon type here
+    pub metadata: Map<String, Value>,
+    pub center: Option<[f64; 2]>,
+    // pub center: Option<[f64; 2]>, // TODO: Use LatLon type here
     pub zoom: Option<f64>,
+    pub bearing: Option<f64>,
     pub pitch: Option<f64>,
+    pub light: Option<Light>,
+    pub terrain: Option<Map<String, Value>>,
+    pub sources: HashMap<String, Source>,
+    pub sprite: Option<String>,
+    pub glyphs: Option<String>,
+    pub transition: Option<Transition>,
+
+    // TODO is this in the style spec?
+    pub projection: Option<Map<String, Value>>,
+
+    pub layers: Vec<StyleLayer>,
+
+    // TODO this is metadata, and it's not in the spec
+    pub created: Option<String>,
+    pub modified: Option<String>,
+    pub id: Option<String>,
+    pub owner: Option<String>,
+    pub visibility: Option<String>,
+    pub protected: Option<bool>,
+    pub draft: Option<bool>,
+
+    // to allow for extra fields in the style
+    #[serde(flatten)]
+    pub extra: Option<Map<String, Value>>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub struct Metadata {
+    Value: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub struct Light {
+    #[serde(default)]
+    anchor: Anchor,
+    #[serde(default)]
+    color: Color,
+    #[serde(default)]
+    intensity: Intensity,
+    #[serde(default)]
+    position: Position,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+struct Color(String);
+impl Default for Color {
+    fn default() -> Self {
+        Color("#ffffff".parse().unwrap())
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+struct Intensity(Vec<f32>);
+impl Default for Intensity {
+    fn default() -> Self {
+        Intensity(vec![1.15, 210.0, 30.0])
```

Comment:
It would be super awesome to link to the maplibre style spec e.g. https://maplibre.org/maplibre-style/layers/#minzoom

This is actually true for all properties. They should preferably link to the docs.